crates/interpreter/src/exec.rs

```diff
@@ -1043,8 +1050,15 @@ impl<'m> Thread<'m> {
         }
         let values = core::mem::take(self.values());
         let frame = self.frame();
-        let Label { arity, kind, .. } = frame.labels.drain(i ..).next().unwrap();
```

Comment:
@ia0 Could you clarify in this line (1046) in the original code the reason to call `drain(i ..)` instead of `drain(l ..)`? `l` is the label index. Thanks.

Reply:
Because `l` starts counting from the end of the vector and `drain()` expected an index from the start of the vector. If Rust had some Python syntax with negative indices (which would probably be a bad idea), then it could have been `drain(-l ..)` (or `-l -1` depending of the actual semantics).

Reply:
Is it possible to give a conceptual explanation on why it is natural for `l` to be indexed reversely?

Reply:
This is WASM specification. They probably did that because it permits copy pasting code without having to change all indices, because they don't depend on their label depth.